src/Conversion/ONNXToTOSA/Math/Conv2D.cpp

```diff
@@ -85,12 +80,60 @@ class ONNXConvOpLoweringToTOSA : public OpConversionPattern<ONNXConvOp> {
           {newPadVec[0], newPadVec[2], newPadVec[1], newPadVec[3]});
     }
 
-    Type newConvOutputType = RankedTensorType::get(
-        {-1, -1, -1, -1}, resultType.cast<ShapedType>().getElementType());
+    Value conv2D = NULL;
+    if (group.getSInt() == 1) {
+      Type newConvOutputType = RankedTensorType::get(
+          {-1, -1, -1, -1}, resultType.cast<ShapedType>().getElementType());
 
-    Value conv2D = tosa::CreateOpAndInfer<tosa::Conv2DOp>(rewriter,
-        op->getLoc(), newConvOutputType, newInput, newWeight, bias, pads,
-        strides, dilations);
+      conv2D = tosa::CreateOpAndInfer<tosa::Conv2DOp>(rewriter, op->getLoc(),
+          newConvOutputType, newInput, newWeight, bias, pads, strides,
+          dilations);
+    } else {
+      // Set up constants outside of loop
+      const int64_t groups = group.getSInt();
+      const int64_t sizeOfSlice = weightShape[1];
```

Comment:
Quite a general question but what happens if `group` is not a divisor of C ? (i.e for instance here, `weightShape[1] * group != C`). Should it be checked or is ONNX mlir already checking that, or is it legal?

Reply:
This did happen when I wrote the test. The verifier of the onnx conv emits an error if the shapes and groups are not compatible.